ou_dedetai/tui_app.py

```diff
@@ -162,63 +176,117 @@ def active_screen(self, value: tui_screen.Screen):
         self._active_screen = value
 
     @property
-    def menu_screen(self) -> tui_screen.MenuScreen:
-        if self._menu_screen is None:
-            self._menu_screen = tui_screen.MenuScreen(
+    def main_screen(self) -> tui_screen.MenuScreen:
+        if self._main_screen is None:
+            self._main_screen = tui_screen.MenuScreen(
                 self,
                 0,
                 self.status_q,
                 self.status_e,
                 "Main Menu",
                 self.set_tui_menu_options(),
             )  # noqa: E501
-        return self._menu_screen
+        return self._main_screen
     
     @property
     def console(self) -> tui_screen.ConsoleScreen:
         if self._console is None:
             self._console = tui_screen.ConsoleScreen(
-                self, 0, self.status_q, self.status_e, self.title, self.subtitle, 0
+                self, 0, self.status_q, self.status_e, 0
             )  # noqa: E501
         return self._console
 
+    @property
+    def header(self) -> tui_screen.HeaderScreen:
+        if self._header is None:
+            self._header = tui_screen.HeaderScreen(
+                self, 0, self.status_q, self.status_e, self.title, self.subtitle, 0
+            )  # noqa: E501
+        return self._header
+
+    @property
+    def footer(self) -> tui_screen.FooterScreen:
+        if self._footer is None:
+            self._footer = tui_screen.FooterScreen(
+                self, 0, self.status_q, self.status_e, 0
+            )  # noqa: E501
+        return self._footer
+
     @property
     def recent_console_log(self) -> list[str]:
         """Outputs console log trimmed by the maximum length"""
         return self.console_log[-self.console_log_lines:]
 
-    def set_window_dimensions(self):
-        self.update_tty_dimensions()
-        curses.resizeterm(self.window_height, self.window_width)
-        self.main_window_ratio = 0.25
+    def set_header_window_dimensions(self):
+        self.header_window_height_min = 2
+        self.header_window_height = min(max(
+            int(self.window_height * self.header_window_ratio), self.header_window_height_min
+        ), 4)
```

Comment:
nit: why is this dynamic? When it's set to static when resizing space doesn't appear between the welcome and the console. Feels odd to have this dynamic
```suggestion
        self.header_window_height = 2
```

Reply:
This is dynamic because the title lines can wrap, causing the title to be as much as 4 lines.

Reply:
This brings up a good question though: when the screen is quite narrow, the title lines wrap. If the screen is small enough, the second title line is hidden because it needs at least 3 lines to display the first part of the subtitle. But in a small enough screen, the header window is only 2 lines. What approach should we take to handle this?

Reply:
In a small enough window (width-wise), the height of the terminal would need to be a minimum of 40 to display the contents of a wrapped TUI subtitle.

Reply:
We could set this to a constant value of 4, but it makes for some funny white space at certain sizes.

Reply:
Are you talking about when the window is this small?
```
  Welcome to Ou Dedetai
  4.0.0-beta.9 (stable)
  Logos Version: 33.1.0.6
  (stable)
```

Reply:
Correct, or thereabout

Reply:
Do we really care about all that information if the screens that small?

Reply:
Of course not, but I wasn't intending to make this into a responsive refactor with properly hiding things.

Reply:
Could we hide the last two lines on small screens? To keep it 2 lines always? Simplify things?

Reply:
Hmm... My preferred way of doing that requires a major refactor. But there is an easy way. Let me test something for it.